src/ttrss_api.cpp

```diff
@@ -161,6 +161,35 @@ json ttrss_api::run_op(const std::string& op,
 	return content;
 }
 
+tagged_feedurl ttrss_api::feed_from_json(const json& jfeed, std::function<std::vector<std::string>(const json&)> tagger) {
+	const int feed_id            = jfeed["id"];
+	const std::string feed_title = jfeed["title"];
+	const std::string feed_url   = jfeed["feed_url"];
+
+	std::vector<std::string> tags;
+	tags.push_back(std::string("~") + feed_title);
+
+	for(auto& tag : tagger(jfeed)) {
```

Comment:
If you don't plan  `tagger` to ever return more than one tag, I'd rather see it replaced by a simple `std::string`, defaulted to an empty value. Current implementations of `tagger` can be written in-place in the loops from which you call `feed_from_json` (see comments below).

Reply:
Based on your comment, I've implemented both variants (receiving a vector as well as a primitive string) as separate commits within this branch.
While I do not have immediate plans to expand the implementation by adding additional automatic tags, one should note that TTRSS itself has tags (which are currently disregarded), which would most likely be added via this codepath.
Looking at both implementations, I would argue that the vector variant is clean enough, while maintaining easier extendability - at a slight runtime cost for constructing the vector.

Reply:
Agreed—vector version is clean enough.